src/material/stepper/_stepper-theme.scss

```diff
@@ -86,24 +86,17 @@
 
   .mat-step-label {
     font: {
-      size: mat-font-size($config, body-1);
-      weight: mat-font-weight($config, body-1);
+      size: mat-font-size($config, body-2);
+      weight: normal;
     };
   }
 
   .mat-step-sub-label-error {
     font-weight: normal;
   }
 
-  .mat-step-label-error {
-    font-size: mat-font-size($config, body-2);
-  }
-
   .mat-step-label-selected {
-    font: {
-      size: mat-font-size($config, body-2);
-      weight: mat-font-weight($config, body-2);
-    };
+    font-weight: bolder;
```

Comment:
Testing this out on the docs site, it seems like `bolder` is a bit thicker than the `500` we had here before. I'm guessing that'll cause some screenshot failures on the presubmit.